types/evidence.go

```diff
@@ -554,6 +555,76 @@ func LightClientAttackEvidenceFromProto(lpb *tmproto.LightClientAttackEvidence)
 // EvidenceList is a list of Evidence. Evidences is not a word.
 type EvidenceList []Evidence
 
+// StringIndented returns a string representation of the evidence.
+func (data *EvidenceList) StringIndented(indent string) string {
+	if data == nil {
+		return "nil-Evidence"
+	}
+	evStrings := make([]string, tmmath.MinInt(len(*data), 21))
+	for i, ev := range *data {
+		if i == 20 {
+			evStrings[i] = fmt.Sprintf("... (%v total)", len(*data))
+			break
+		}
+		evStrings[i] = fmt.Sprintf("Evidence:%v", ev)
+	}
+	return fmt.Sprintf(`EvidenceList{
+%s  %v
+%s}#%v`,
+		indent, strings.Join(evStrings, "\n"+indent+"  "),
+		indent, data.Hash())
+}
+
+// ByteSize returns the total byte size of all the evidence
+func (data *EvidenceList) ByteSize() int64 {
+	if len(*data) != 0 {
+		pb, err := data.ToProto()
+		if err != nil {
+			panic(err)
+		}
+		return int64(pb.Size())
+	}
+	return 0
+}
+
+// FromProto sets a protobuf EvidenceList to the given pointer.
+func (data *EvidenceList) FromProto(eviData *tmproto.EvidenceList) error {
+	if eviData == nil {
+		return errors.New("nil evidence list")
```

Comment:
shouldn't we just let this panic? I think nil checking everything may be safe but also maybe worth avoiding these potential situations

Reply:
I don't mind changing it but this was/is the style in many FromProto/ToProto functions and wherever these things are called, error handling is in place.

Reply:
yeah, I can see that. It's fine to leave it this way, so that things match, but I think it's worth calling out that this isn't exactly the best pattern!

Reply:
For protobufs and types adjacent to protobufs, I think this pattern is OK—the proto compilers all use `nil` dispatch to simplify use of "unset" fields. 

Doing this in a non-proto-adjacent type would definitely want some documentation, but here I think it's actually fine.